src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTriggerTest.java

```diff
@@ -0,0 +1,189 @@
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.anyInt;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import hudson.model.BooleanParameterDefinition;
+import hudson.model.FileParameterDefinition;
+import hudson.model.FreeStyleProject;
+import hudson.model.ParameterDefinition;
+import hudson.model.ParameterValue;
+import hudson.model.ParametersAction;
+import hudson.model.ParametersDefinitionProperty;
+import hudson.model.StringParameterDefinition;
+import hudson.model.StringParameterValue;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.MockitoRule;
+import org.mockito.quality.Strictness;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StashBuildTriggerTest {
+
+  @Rule public JenkinsRule jenkinsRule = new JenkinsRule();
+  @Rule public MockitoRule rule = MockitoJUnit.rule().strictness(Strictness.STRICT_STUBS);
+
+  private StashBuildTrigger trigger;
+
+  private StashCause cause;
+  private FreeStyleProject project;
+
+  @Mock private ParametersDefinitionProperty parametersDefinitionProperty;
+
+  @Before
+  public void before() throws Exception {
+    trigger =
+        new StashBuildTrigger(
+            "ProjectPath",
+            "* * * * *",
+            "StashHost",
+            "CredentialsId",
+            "ProjectCode",
+            "RepositoryName",
+            "CiSkipPhrases",
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            "CiBuildPhrases",
+            false,
+            "TargetBranchesToBuild",
+            false);
+
+    project = spy(jenkinsRule.createFreeStyleProject());
+    trigger.start(project, true);
+  }
+
+  private StashCause makeCause(Map<String, String> additionalParameters) {
+    return new StashCause(
+        "StashHost",
+        "SourceBranch",
+        "TargetBranch",
+        "SourceRepositoryOwner",
+        "SourceRepositoryName",
+        "PullRequestId",
+        "DestinationRepositoryOwner",
+        "DestinationRepositoryName",
+        "PullRequestTitle",
+        "SourceCommitHash",
+        "DestinationCommitHash",
+        "BuildStartCommentId",
+        "PullRequestVersion",
+        additionalParameters);
+  }
+
+  private void jobSetup(List<ParameterDefinition> parameterDefinitions) {
```

Comment:
varargs would make this even easier to use I think

Reply:
Good idea. Created #98.